client/src/utils/fight/updateWeapons.ts

```diff
@@ -20,13 +20,13 @@ const updateWeapons = (
     throw new Error('Spritesheet not found');
   }
 
-  // Only affect the UI for main brutes
-  if (brute.bust) {
+  // Only affect the HUD for the focused brutes
+  if (brute.HUDFocused) {
     // Empty list
     brute.weaponsIllustrations.forEach((illustration) => {
       illustration.destroy();
     });
-    brute.weaponsIllustrations = [];
+    brute.weaponsIllustrations.splice(0);
```

Comment:
Why this change?

splice returns the spliced elements, meaning more memory assigned, where simply assigning a new array doesn't.
As the illustration references are destroyed above, assigning a new array does not create a memory leak.

Reply:
This change was made because the weaponIllustration array is shared with every brute. Should I 
1) Rename it "teamWeaponIllustrations" ? 
2) Or call a new method "destroyWeapons" for previously displayed brute, then updateWeapons for newly displayed one ?

Reply:
I don't understand your logic.
splice(0) does the same thing as  = [], as it empties the array.

It doesn't change anything other than waste memory.

Reply:
= [] creates a new empty array, whereas splice(0) empty the array, this is not the same thing

Here, I need brute.weaponsIllustrations array to keep it's pointer as this array is in fact the team weaponIllustration array, which is emptied and filled everytime a new brute is displayed.

Therefore, either you are ok with just renaming it "teamWeaponIllustration" for clarity, or I make weaponIllustration not shared anymore but I would have to add a new "removeWeaponsHUD" function or add a "previousDisplayedBrute" agrument to updateWeapons.

Reply:
I could also do it like this :

while (brute.weaponsIllustrations.length){

  brute.weaponsIllustrations.pop().destroy();
  
}

But I don't know how ok you are with while loops

Reply:
I didn't realize everyone on the team shared the same array reference.
In this case, renaming it is better.
Also, don't use splice, use .length = 0 to clear it instead.